Added the 6th and last chapter of Nest gs (#8)
raven-os/getting-started: nest/06.glossary.md

# 6 Glossary

## A

## B

## C
* Cache

In Nest, the cache is a local copy of the list of available packages from Raven-OS server.
The cache has to be kept up-to-date in order to install the latest versions of the packages.

* Category

> Inside each repository, there are several categories.

Reviewer: Cross-link on repository?

Author: good idea
and yes I totally forgot to link the terms to their relevant section, it was originally planned lel

> The packages are split between the repositories and their categories.

Reviewer: Cross-link on packages?

Each package name must be unique in a category of a repository, but there can be several packages with the same name spread in several categories.

## D
* Dependency

> There is a dependency when a package depends on another one to function properly.

Reviewer: Unclear. The phrase should be of the form "a dependency is" I think?

* Dependency graph/Dependency tree

> The dependency graph is a directed graph representing the dependencies of packages.

Reviewer: ```suggestion
The dependency graph is a directed graph (i.e. a graph with directed connections) representing the dependencies between packages.
```

## E

## F
* `finest`

`finest` is the binary of Nest targeted towards experienced users.

* `finest group create <name> [--parent <group>]`

The command of `finest` to create a group, with `@root` as a parent by default.

* `finest group list`

The command of `finest` to list all groups on the OS.

* `finest group remove <name> [--parent <group>]`

The command of `finest` to remove a group, with `@root` as a parent by default.

* `finest merge`

The command of `finest` to merge the dependency graph `finest` altered with the transactions you applied, with the dependency graph currently in use by the OS.

* `finest requirement add <requirement> [--parent <group>]`

The comand of `finest` to add a requirement to a group, by default `@root`.

* `finest requirement remove <requirement> [--parent <group>]`

The command of `finest` to remove a requirement from a group, by default `@root`.

## G
* Groups

> Groups are a way to put packages together, to handle their installation and uninstallation in a more efficient manner.

Reviewer: ```suggestion
Groups are a way to group packages together, in order to install and uninstall them in a more efficient manner.
```
(I know you used "put" to avoid saying "group" but you are defining groups *in the context of nest*, so referring to grouping things is not a problem.)

## H

## I

## J

## K

## L

## M
* Mirror

A mirror is a server with a copy of all the data from a main server.

## N
* Nest

Nest is a dependency-based package manager for Raven-OS.
It consists of two binaries: `nest` and `finest`.

* `nest`

The binary of Nest mainly targeted towards newcomers.

* `nest install <package>`

The command of `nest` to install a package.

* `nest list`

The command of `nest` to list all installed packages.

* `nest pull`

The command of `nest` to update the list of available packages that you can install with Nest.

* `nest uninstall <package>`

The command of `nest` to uninstall a package.

## O

## P
* Package

A package is a compressed file archive containing all the files required to install, uninstall and update a particular software or application.

* Package manager

A package manager allows the user to install, upgrade or remove software packages on their OS, in other words, to manage packages.

## Q

## R
* Repository

A repository contains packages.
In each repository used by Nest, there are several categories.

> * Requirement

Reviewer: Difference with "dependency" ?

The cache has to be kept up-to-date in order to install the latest versions of the packages.

* Category

[//]: # (TODO: add links to the 'repository' and 'package' entry of this glossary)
Inside each [repository](), there are several categories.
The [packages]() are split between the repositories and their categories.
Each package name must be unique in a category of a repository, but there can be several packages with the same name spread in several categories.

## D
[//]: # (TODO: add link to the section 1.2)
* Dependency [section 1.2]()

[//]: # (TODO: add link to the 'package' entry of this glossary)
A dependency is a [package]() A that is required by a package B, so that the package B can function properly.

[//]: # (TODO: add link to the section 4.1.3)
* Dependency graph/Dependency tree [section 4.1.3]()

[//]: # (TODO: add link to the 'package' entry of this glossary)
The dependency graph is a directed graph (i.e. a graph with directed connections) representing the dependencies between [packages]().
## E

## F
[//]: # (TODO: add link to the chapter 4)
* `finest` [chapter 4]()

`finest` is the binary of Nest targeted towards experienced users.

> * `finest group create <name> [--parent <group>]`

Reviewer 2: I'm not sure commands should be in the glossary. What was your idea behind that?

Author: the glossary seemed empty, so I wanted to add more things to it D:
but I agree, they shouldn't be there, I'll remove them

The command of `finest` to list all [groups]() on the OS.

* `finest group remove <name> [--parent <group>]`

[//]: # (TODO: add link to the 'group' entry of this glossary)
The command of `finest` to remove a [group](), with `@root` as a parent by default.

* `finest merge`

[//]: # (TODO: add link to the 'Dependency graph' and 'transaction' entries of this glossary)
The command of `finest` to merge the [dependency graph]() `finest` altered with the [transactions]() you applied, with the dependency graph currently in use by the OS.

* `finest requirement add <requirement> [--parent <group>]`

[//]: # (TODO: add link to the 'requirement' and 'group' entries of this glossary)
The comand of `finest` to add a [requirement]() to a [group](), by default `@root`.

* `finest requirement remove <requirement> [--parent <group>]`

[//]: # (TODO: add link to the 'requirement' and 'group' entries of this glossary)
The command of `finest` to remove a [requirement]() from a [group](), by default `@root`.

## G
[//]: # (TODO: add link to the section 4.1.4)
* Groups [section 4.1.4]()

[//]: # (TODO: add link to the 'package' entry of this glossary)

> Groups are a way to group [packages]() together, in order to install and uninstall them in a more efficient manner.

Reviewer 2: ```suggestion
Groups are a way to organize and regroup [packages]() together, in order to install and uninstall them in a more efficient manner.
```

Reviewer: Apparently "regroup" is valid English. I haven't heard it often though.

## H

## I

## J

## K

## L

## M
* Mirror

> A mirror is a server with a copy of all the data from a main server.

Reviewer 2: ```suggestion
A mirror is an alternative server with a copy of all the data from a main server. They are used when the main server is out of service.
```

Reviewer: If you are going that direction, maybe also mention that they can speed up download speed by being closer geographically.

[//]: # (TODO: add link to the 'package' entry of this glossary)
The command of `nest` to install a [package]().

* `nest list`

[//]: # (TODO: add link to the 'package' entry of this glossary)
The command of `nest` to list all installed [packages].

* `nest pull`

[//]: # (TODO: add link to the 'package' entry of this glossary)
The command of `nest` to update the list of available [packages] that you can install with Nest.

* `nest uninstall <package>`

The command of `nest` to uninstall a package.

## O

## P
[//]: # (TODO: add link to the section 1.1)
* Package [section 1.1]()

A package is a compressed file archive containing all the files required to install, uninstall and update a particular software or application.

[//]: # (TODO: add link to the section 1.1)
* Package manager [section 1.1]()

[//]: # (TODO: add link to the 'package' entry of this glossary)
A package manager allows the user to install, upgrade or remove software [packages]() on their OS, in other words, to manage packages.

## Q

## R
[//]: # (TODO: add link to the section 1.3)
* Repository [section 1.3]()

[//]: # (TODO: add link to the 'package' and 'category' entries of this glossary)

> A repository contains [packages].

Reviewer 2: ```suggestion
A repository is a collection of [packages] maintained by the same person or organisation, with all [packages]() usually following with the same policy and rules.
```

Reviewer: ```suggestion
A repository is a collection of [packages] maintained by the same person or organisation, with all [packages]() usually following the same policy and rules.
```
(you put a "with" to much, idk if I should edit your comment or add a new one like this)

> In each repository used by Nest, there are several [categories]().

Reviewer 2: ```suggestion
In each repository used by Nest, there are several [categories](), which in turns, contains several [packages]().
```

Reviewer: ```suggestion
In each repository used by Nest, there are several [categories](), which in turn contain several [packages]().
```
"turns contains" -> "turn contain"

* Requirement

[//]: # (TODO: add link to the 'dependency graph', 'package' and 'dependency' entries of this glossary)

> Each node of the [dependency graph]() has a requirement that is fulfilled by another node.

Reviewer 2: Some nodes don't have any requirements. And requirements may not be fulfilled by a package: it may be fulfilled by a group.

For example, when doing `nest group create @foo --parent @root`, the node `@root` depends on `@foo`. Packages can't requires group though, as they may not exist in the user's dependency graph.

A requirement is a [dependency]() that can be fulfilled by one and only one [package](), and is noted as `repository::category/package#[version|version_requirement]`.

## S

> * `stable` repository

Reviewer 2: Maybe describe the `beta` and `unstable` repository too?
